src/Common/src/Interop/Windows/kernel32/Interop.GetFileAttributesEx.cs

```diff
@@ -68,10 +67,8 @@ internal unsafe struct WIN32_FIND_DATA
             internal uint nFileSizeLow;
             internal uint dwReserved0;
             internal uint dwReserved1;
-            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 260)]
-            internal string cFileName;
-            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 14)]
-            internal string cAlternateFileName;
+            internal FixedString.Size260 cFileName;
+            internal FixedString.Size14 cAlternateFileName;
```

Comment:
@jkotas I started with `fixed char foo[n]`, helpers, and properties on the struct. On my own interop library I found that it was easy to make mistakes and difficult to add useful functionality. This pattern evolved out of making repeated, stupid mistakes that were difficult to catch. If we didn't have to use `fixed(char* c = foo)` to get the `char*` out I'd be more amenable to using the "raw" fixed size array, even though it still doesn't address making mistakes in passing the correct size around.

I'm struggling to understand why something that safely handles capacity and makes functionality safe & discoverable is not an improvement. What is the real downside to using these? It isn't more complicated than loose helper methods and sticking consts all over our structs. Additionally, there are only a few sizes Windows actually uses- this pattern isn't open-ended in generating new types.